examples/platforms/nrf528xx/src/radio.c

```diff
@@ -1216,4 +1216,19 @@ void otPlatRadioUpdateCslSampleTime(otInstance *aInstance, uint32_t aCslSampleTi
 
     sCslSampleTime = aCslSampleTime;
 }
+
+#if OPENTHREAD_CONFIG_CSL_USE_DELAY_TRANSMIT
+otError otPlatRadioTransmitAfter(otInstance *aInstance, otRadioFrame *aFrame, uint32_t aDelay)
+{
+    otError error = OT_ERROR_NONE;
+
+    aFrame->mPsdu[-1] = aFrame->mLength;
+
+    error = nrf_802154_transmit_raw_at(&aFrame->mPsdu[-1], false, (uint32_t)nrf5AlarmGetCurrentTime(), aDelay,
```

Comment:
I'm not sure if using `nrf5AlarmGetCurrentTime()` in precise TX function is the best approach. Calling functions like this decreases precision, because it is prone to software latency.

The idea behind `nrf_802154_transmit_raw_at()` is to provide precise TX time. The program flow should be like this:

```
nrf_802154_received_raw(frame, ..., timestamp)
{
  m_rx_frame = frame;
  m_rx_timestamp = timestamp;
}

otPlatRadioTransmitPrecisely(aFrame, ...)
{
  nrf_802154_transmit_raw_at(aFrame, ..., m_rx_timestamp, getCslPhaseInUs(m_rx_frame) + x * getCslPeriodInUs(m_rx_frame), ...);
}
```

The result is precise TX timing, related only to RX timestamp. Software latency has no impact on radio timings with such approach.

Would it be possible to change proposed functions to support code flow similar to what I suggested?